### PR TITLE
feat(streaming_load): Support file_format syntax in streaming load insert sql

### LINKED ISSUE #8604: Feature:  support `format  (type=ndjson ...)`
we can use it in streaming load.

`(type=ndjson ...)` is the same with the in StageFormatOptions.

1. the format related are written close to each other
2. not need http header or url parameter (only to be compatible with clickhouse)   
     1 .  http header can not be checked
     2.  url parameter need escape. https://github.com/datafuselabs/databend/issues/8243
3.  avoid supporting None in Settings https://github.com/datafuselabs/databend/issues/8541

### PR DESCRIPTION
**Pull request by ariesdevil**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Support COPY INTO FILE_FORMAT syntax for the streaming load to unify usage.

Closes #8604

### CODE REVIEW

File: src/query/ast/src/parser/statement.rs [hunk 1]
```diff
@@ -1078,12 +1078,23 @@ pub fn statement(i: Input) -> IResult<StatementMsg> {
 pub fn insert_source(i: Input) -> IResult<InsertSource> {
     let streaming = map(
         rule! {
-            FORMAT ~ #ident ~ #rest_str
-        },
-        |(_, format, (rest_str, start))| InsertSource::Streaming {
-            format: format.name,
-            rest_str,
-            start,
+            FORMAT ~ #ident ~ #rest_str ~ (FILE_FORMAT ~ "=" ~ #options)?
```
Comment (youngsofun) [inline src/query/ast/src/parser/statement.rs:1081]: what about `FORMAT (~ #ident |  "=" ~ #options ) ~ #rest_str ` ? cc @sundy @xuanwo

File: src/query/settings/src/lib.rs [hunk 2]
```diff
@@ -491,6 +492,38 @@ impl Settings {
         }))
     }
 
+    pub fn set_file_format_options(&self, opts: &FileFormatOptions) -> Result<()> {
```
Comment (youngsofun) [inline src/query/settings/src/lib.rs:495]: when we already have FileFormatOptions, should try to avoid using settings (our final goal)

Comment (ariesdevil) [reply src/query/settings/src/lib.rs:495]: This is for compatibility, if we decide to remove the old syntax, this will be removed future.

Comment (youngsofun) [reply src/query/settings/src/lib.rs:495]: we should only support new syntax internally,  and  try to convert old syntax to new.

Comment (youngsofun) [reply src/query/settings/src/lib.rs:495]: FileFormatOptions -> setting is the reverse

File: src/meta/types/src/user_stage.rs [hunk 3]
```diff
@@ -164,6 +182,21 @@ impl FromStr for StageFileFormatType {
     }
 }
 
+impl ToString for StageFileFormatType {
+    fn to_string(&self) -> String {
+        match *self {
```
Comment (youngsofun) [inline src/meta/types/src/user_stage.rs:187]: what about `format!("{:?}", *self)`

File: src/query/service/src/servers/http/clickhouse_handler.rs [hunk 4]
```diff
@@ -290,6 +290,52 @@ pub async fn clickhouse_handler_post(
...
+                .await
+                .map_err(InternalServerError)?;
+
+            // override settings
+            settings
+                .set_file_format_options(option_settings)
+                .map_err(InternalServerError)?;
+
+            let input_context = Arc::new(
+                InputContext::try_create_from_insert(
```
Comment (youngsofun) [inline src/query/service/src/servers/http/clickhouse_handler.rs:323]: we need a new `try_create_from_insert_v2` which do not use settings to pass options.

Comment (youngsofun) [reply src/query/service/src/servers/http/clickhouse_handler.rs:323]: you can refer to try_create_from_copy

File: src/query/settings/src/lib.rs (hunk 2 above)
Comment (youngsofun) [reply src/query/settings/src/lib.rs:495]: we should only support new syntax internally, and try to convert old syntax to new.

it is also ok to keep both code path for now.
but new syntax should not based on the old,  so we need InputContext::try_create_from_insert_v2

File: src/query/service/src/servers/http/clickhouse_handler.rs (hunk 4 above)
Comment (ariesdevil) [reply src/query/service/src/servers/http/clickhouse_handler.rs:323]: good idea, with `try_create_from_insert_v2` we can avoid use settings.

**Comment by BohuTANG** [comment]
Please add more summary about this PR:)

**Review by youngsofun — APPROVED** [review]

**Pull request merged**